lib/cos_agent.py

```diff
@@ -0,0 +1,1427 @@
+# Copyright 2023 Canonical Ltd.
```

Comment:
When using `charmcraft fetch-lib` to get this file, it would be placed in `lib/charms/grafana_agent/v0/cos_agent.py`.

Should we perhaps use the same location so that others find it where they would expect it?

There also needs to be some documentation on how to update the library code.

(From cursory view we cannot run `charmcraft fetch-lib` directly here, which makes sense because this is not a full charm repository which `charmcraft` would expect.)

Reply:
A very dirty way of handling this would be to put "placeholder" charmcraft.yaml with minimal set of required fields for `charmcraft fetch-libs` to work.

Alternatively I was thinking about tracking the cos_agent library via submodules, but if I understand correctly (please correct me if I'm wrong @sed-i ), the library source code doesn't have it's own repository, its source lives in [grafana agent charm](https://github.com/canonical/grafana-agent-operator/tree/main/lib/charms/grafana_agent/v0). So a simple git submodule is not an option, unless we want to pull entire grafana agent charm. Which feels even dirtier that my first proposal.